hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneHAClusterImpl.java

```diff
@@ -966,4 +970,31 @@ public StorageContainerManager getStorageContainerManager() {
     return getStorageContainerManagers().get(0);
   }
 
+  private int getFreePort() {
+    ServerSocket ss = null;
+    try {
+      ss = new ServerSocket(0);
+      return ss.getLocalPort();
+    } catch (IOException e) {
+      e.printStackTrace();
+    } finally {
+      if (ss != null) {
+        try {
+          ss.close();
+        } catch (IOException e) {
+          LOG.error("Got exception while closing ServerSocket: " +
+              e.getMessage());
+        }
+      }
+    }
+    return -1;
+  }
+
+  private Set<Integer> getFreePortSet(int size) {
+    Set<Integer> portSet = new HashSet<>();
+    while (portSet.size() < size) {
+      portSet.add(getFreePort());
```

Comment:
If getFreePort() returns -1, then it will just add it to the set, which will cause problems later I think?

However if we switch to use `SCMTestUtils.getReusableAddress()` then we can handle the exception it throws and try again to get an address.

Reply:
This reminds me: we could use `NetUtils.createLocalServerAddress(int)` from Ratis to get multiple free ports.

https://github.com/apache/ratis/blob/5e444084307dce717816dd16c7942d785fd3e9d1/ratis-common/src/main/java/org/apache/ratis/util/NetUtils.java#L119-L143

Reply:
Sure, updated to use `SCMTestUtils.getReusableAddress()`.

Reply:
The Ratis version is nice. It actually binds the socket and keeps it bound while it gets the rest of the requested sockets. That means it will return a list of unique sockets for sure, while the version we are working with, could return duplicate ports, although that is being handled via adding the port to the set.

Reply:
Yes, the ratis version is more mature. Have updated to use the ratis version.
Got a small problem, if I didn't use ratis in the project, but I'd want to use the utils provided by ratis, should I copy the code or add the dependency?

Reply:
As we already have a dependency on Ratis across the project its fine to add the dependency. If we didn't I probably would copy the code, rather than take a dependency just for this.